subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -783,7 +783,7 @@ static int gatt_proxy_advertise(void)
 {
 	int err;
 
-	int32_t max_adv_duration;
+	int32_t max_adv_duration = 0;
```

Comment:
Isn't the core issue that `active_proxy_sub_cnt_get()` has a signed return value, even though it will always return `>= 0`. This leads to the compiler (rightly) thinking that the if-else if construction not covering the `< 0` case where `max_adv_duration` would indeed not be initialized.

Reply:
Good point. I also missed the conditional branch at line 839.

Reply:
That said, I hope the Mesh maintainers take note that there are things which could be improved/cleaned up with this code.